kinto_attachment/tests/test_views_attachment.py

```diff
@@ -146,6 +164,52 @@ def test_upload_refused_if_extension_not_allowed(self):
         self.assertEqual(resp.json['message'],
                          'body: File extension is not allowed.')
 
+    # Permissions.
+
+    def test_upload_refused_if_not_authenticated(self):
+        self.headers.pop('Authorization')
+        self.upload(status=401)
+
+    def test_upload_refused_if_not_allowed(self):
+        self.headers.update(get_user_headers('jean-louis'))
+        self.upload(status=403)
+
+    def test_upload_replace_refused_if_only_create_allowed(self):
```

Comment:
Why? I think if you can create a record, you should be able to upload the file as well.

This is because then you will get the write permission on the record anyway.